Removing dependency from XML files and creating them as python scripts

siddharth7113 opened the pull request:
## PR details

- Currently MAWS require XML file to be loaded as `Structure` object while selecting aptamer type, but this can be done in-memory through `DNA.py` and `RNA.py` and can remove parsing the XML object.
- Also added type hints to`Structure.py`  to help in understanding code better (AI generated mostly but  verified once) 
- Tested the algorithm and generation works with `DNA.py` and `RNA.py`.

fkiraly reviewed (approved):
Very nice, localized refactor.

This makes the code - and the workflow - substantially easier to understand, and it makes the logic seriously easier to inspect.

Non-blocking suggestions:
* Perhaps you could also add an example to the `Structure` object docstring, even if it is very long (such as, the code from `rna_structure`, or a simplified version thereof).
* to make it clearer that `rna_structure` is actually a hard-coded example, perhaps rename it? `load_example_rna_structure` or similar?

Regarding where to get the python `Structure` populated from, your refactor clarifies that we have a missing link here between raw `pdb` and the content from `rna_structure` - maybe pybiomed is an interesting candidate? https://github.com/gc-os-ai/pyaptamer/issues/120

siddharth7113 commented:
> Very nice, localized refactor.
> 
> This makes the code - and the workflow - substantially easier to understand, and it makes the logic seriously easier to inspect.
> 
> Non-blocking suggestions:
> 
> * Perhaps you could also add an example to the `Structure` object docstring, even if it is very long (such as, the code from `rna_structure`, or a simplified version thereof).
> * to make it clearer that `rna_structure` is actually a hard-coded example, perhaps rename it? `load_example_rna_structure` or similar?
> 
> Regarding where to get the python `Structure` populated from, your refactor clarifies that we have a missing link here between raw `pdb` and the content from `rna_structure` - maybe pybiomed is an interesting candidate? [gc-os-ai/pyaptamer#120](https://github.com/gc-os-ai/pyaptamer/issues/120)

Sure, I'll add the docstring for `Structure` object  and look into `pybiomed` whether it is suitable for our purposes.

siddharth7113 commented:
@fkiraly  I have rename  and deleted files not needed and added docstrings for Structure.py , this PR has already became too big, so I am thinking if it's good we can go ahead and merge it, and as a stack I can add few things such as pyproject and linters so that we can run tests on it.

siddharth7113 commented:
> Happy with the change - do I understand correctly that the `xml` are replaced by two hard coded `Structure` instances?
> 
> If so, that's great, I would just rename `rna_structure` to `load_maws_rna_example` or similar.

Yes, that is correct! , I  will make these changes.
